### PR TITLE
Garbage input to setter fix

### PR DESCRIPTION
**Pull request by mbad0la**
**Work in Progress**

Hi @mj1856 @maggiepint 

This PR is in reference to #3483 

Please review the current changes and let me know if this hack seems to be doing the trick. Also, let me know of any other steps I need to take to make this PR ready for a merge. Thanks :)

### CODE REVIEW

**Comment by mj1856** [comment]
Sorry, this isn't good enough.  It needs to actually test the `value` input parameter of the [`set`](https://github.com/moment/moment/blob/2.15.1/src/lib/moment/get-set.js#L24) function.

**Comment by mbad0la** [comment]
So basically, You don't want the setter to set the illegal value and then check whether the instance is valid or not right? You want to prompt at the setter itself, by checking the value?

**Comment by maggiepint** [comment]
Correct.

**Comment by mbad0la** [comment]
Just a minor doubt. Is there a method in moment to check for the validity of an input as an year, month etc.?

**Comment by maggiepint** [comment]
It really need only be numeric. The setters will all overflow into the next
unit if they are out of range. That behavior is expected. So the only thing
that needs validation is that the input is a number, or a string that can
be parsed to a number.

**Comment by mbad0la** [comment]
Hi! I've identified where to change the code. Just need to know how do you need the rejection of illegal input to be handled. Silently ignore it or do something else? Please let me know @mj1856 @maggiepint 

Thanks :)

**Comment by ichernev** [comment]
@mbad0la like everything else in moment -- we silently ignore errors. If its the constructor then you can make the moment invalid. If its a setter, just ignore the error.

**Comment by mbad0la** [comment]
@maggiepint 
> It really need only be numeric. The setters will all overflow into the next
unit if they are out of range. That behavior is expected. So the only thing
that needs validation is that the input is a number, or a string that can
be parsed to a number.

I used this information and added a `NaN` check on the setter but that has broken the module. It seems the setter does have input types other than numerical. Can you please take a look at it?

File: src/lib/create/valid.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import getParsingFlags from '../create/parsing-flags';
 import some from '../utils/some';
 
 export function isValid(m) {
-    if (m._isValid == null) {
+    if (m._isValid === null || m._isValid === true) {
```
Comment (maggiepint) [inline src/lib/create/valid.js]: This changes validity evaluation from one to continual. I don't see why this change would need to cause that.

**Comment by maggiepint** [comment]
Closing in favor of #3704
